staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/testing/testserver.go

```diff
@@ -109,11 +111,19 @@ func StartTestServer(t Logger, instanceOptions *TestServerInstanceOptions, custo
 	s := options.NewCustomResourceDefinitionsServerOptions(os.Stdout, os.Stderr)
 	s.AddFlags(fs)
 
-	s.RecommendedOptions.SecureServing.Listener, s.RecommendedOptions.SecureServing.BindPort, err = createListenerOnFreePort()
+	s.RecommendedOptions.SecureServing.Listener, s.RecommendedOptions.SecureServing.BindPort, err = createLocalhostListenerOnFreePort()
 	if err != nil {
 		return result, fmt.Errorf("failed to create listener: %v", err)
 	}
 	s.RecommendedOptions.SecureServing.ServerCert.CertDirectory = result.TmpDir
+	s.RecommendedOptions.SecureServing.ExternalAddress = s.RecommendedOptions.SecureServing.Listener.Addr().(*net.TCPAddr).IP // use listener addr although it is a loopback device
+
+	_, thisFile, _, ok := runtime.Caller(0)
```

Comment:
Is there an alternative to this? What does this assume about the working directory of the test?

Reply:
it is the positions of this file, i.e. the fixtures are next to it. This avoid that we need `n` fixtures for  `n` test packages using this test server.